### PR TITLE
fix gc config

### PR DESCRIPTION
**Pull request by Maoni0**
fix the issue introduced by https://github.com/dotnet/coreclr/pull/25574

### CODE REVIEW

**Comment by Maoni0** [comment]
unfortunately with https://github.com/dotnet/coreclr/pull/25574 we were no longer reading from the env var, only from the runtime config. sorry I neglected to notice it when did the CR.

File: src/vm/eeconfig.cpp [hunk 1]
```diff
@@ -808,18 +810,20 @@ fTrackDynamicMethodDebugInfo = CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_
...
+    iGCHeapHardLimit = GetConfigDWORD_DontUse_(CLRConfig::EXTERNAL_GCHeapHardLimit, iGCHeapHardLimit);
+    if (!iGCHeapHardLimit) iGCHeapHardLimit = Configuration::GetKnobDWORDValue(W("System.GC.HeapHardLimit"));
 #endif //_WIN64
 
-    const ULONGLONG ullHeapHardLimit = Configuration::GetKnobULONGLONGValue(W("System.GC.HeapHardLimit"));
-    iGCHeapHardLimit = FitsIn<size_t, ULONGLONG>(ullHeapHardLimit)
-        ? static_cast<size_t>(ullHeapHardLimit)
-        : ClrSafeInt<size_t>::MaxInt();
-    iGCHeapHardLimitPercent = Configuration::GetKnobDWORDValue(W("System.GC.HeapHardLimitPercent"), 0);
+    iGCHeapHardLimitPercent = GetConfigDWORD_DontUse_(CLRConfig::EXTERNAL_GCHeapHardLimitPercent, iGCHeapHardLimitPercent);
```
Comment (sergiy-k) [inline src/vm/eeconfig.cpp]: Just curious: can this be done using one line call ```Configuration::GetKnobDWORDValue(W("System.GC.HeapHardLimitPercent"), CLRConfig::EXTERNAL_GCHeapHardLimitPercent)```

I.e. similar to how it is done for ```System.GC.LOHThreshold``` a few lines below. Or is this switch special in some way?

Comment (Maoni0) [reply src/vm/eeconfig.cpp]: no, because it's of a different type from the LOHThreshold config and GetKnobDWORDValue doesn't take it (I already tried).

**Comment by mjsabby** [comment]
Thanks @maoni0. @sudeepagarwal This fixes #25889

File: src/vm/eeconfig.cpp (hunk 1 above)
Comment (sergiy-k) [reply src/vm/eeconfig.cpp]: Thank you!

**Comment by MeiChin-Tsai** [comment]
Approved for .NET Core 3.0. Thanks.

**Pull request merged**